src/components/Facility/FacilityForm.tsx

```diff
@@ -45,11 +45,14 @@ import { Organization } from "@/types/organization/organization";
 
 interface FacilityProps {
   organizationId?: string;
+  navOrganizationId?: string;
```

Comment:
No, you should be using the organizationId, not navOrganizationId. Latter refers to the parent at level 1 (i.e. district level).